libcxx/test/std/ranges/range.adaptors/range.enumerate/iterator/base.pass.cpp

```diff
@@ -0,0 +1,81 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17, c++20
+
+// <ranges>
+
+// class enumerate_view
+
+// class enumerate_view::iterator
+
+// constexpr const iterator_t<Base>& base() const & noexcept;
+// constexpr iterator_t<Base> base() &&;
+
+#include <ranges>
+
+#include <array>
+#include <cassert>
+#include <concepts>
+#include <utility>
+#include <tuple>
+
+#include "test_iterators.h"
+#include "../types.h"
+
+template <class Iterator>
+constexpr void testBase() {
+  using Sentinel          = sentinel_wrapper<Iterator>;
+  using View              = MinimalView<Iterator, Sentinel>;
+  using EnumerateView     = std::ranges::enumerate_view<View>;
+  using EnumerateIterator = std::ranges::iterator_t<EnumerateView>;
+
+  auto make_enumerate_view = [](auto begin, auto end) {
+    View view{Iterator(begin), Sentinel(Iterator(end))};
+
+    return EnumerateView(std::move(view));
+  };
+
+  std::array array{0, 1, 2, 3, 84};
+  const auto view = make_enumerate_view(array.begin(), array.end());
+
+  // Test the const& version
+  {
+    EnumerateIterator const it                          = view.begin();
+    std::same_as<const Iterator&> decltype(auto) result = it.base();
+    ASSERT_NOEXCEPT(it.base());
+    assert(base(result) == array.begin());
+  }
+
+  // Test the && version
+  {
+    EnumerateIterator it                         = view.begin();
```

Comment:
Copying comments from Phabricator:
> Can we check that the iterator was moved?

What I had in mind was to define a simple iterator class that has a member variable like `bool was_moved_ = false;` that is set to `true` in the move constructor (very similar to what you do in `base.pass.cpp` for the view itself).